build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -532,14 +532,18 @@ public static String querySystemPackages(IExecutionEnvironment environment, Prop
 				if (javaPackages == null) {
 					String profileSystemPackages = preJava9ProfileProperties.getProperty(Constants.FRAMEWORK_SYSTEMPACKAGES, ""); //$NON-NLS-1$
 					if (profileSystemPackages.isBlank()) {
-						return null;
+						return List.of();
 					}
 					javaPackages = COMMA.splitAsStream(profileSystemPackages).filter(p -> p.startsWith("java.")).toList(); //$NON-NLS-1$
 				}
 				IVMInstall targetVM = JavaRuntime.getDefaultVMInstall(); // Set by the Target-Definition if specified there
+				if (targetVM == null) {
+					LOGGER.warn("No default JRE installation selected"); //$NON-NLS-1$
+					return List.of();
+				}
 				Collection<String> targetVMSystemPackages = querySystemPackages(targetVM, null);
```

Comment:
I would rather let querySystemPackages return a stream instead instead it is even converted to a stream right after that... and having `null` as return value for collections is really ugly....